### PR TITLE
tf 2.5, keras functional api, tf.data.Dataset

### PR DESCRIPTION
**Pull request by max1mn**
Hello, this PR adds tensorflow 2.5.0 support with keras functional api. Also, input_data can be packed (input, labels) tf.data.Dataset. 

tf.data.Dataset example:
tf_dataset = tf.data.Dataset.from_generator(generator_func, output_types=output_types, output_shapes=output_shapes)
tf_dataset = tf_dataset.batch(16)
grads = get_gradients(model, 1, tf_dataset, None)

### CODE REVIEW

**Comment by OverLordGoldDragon** [comment]
Thanks for the PR.
